apis/placement/v1beta1/override_types.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",categories={fleet,fleet-placement}
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Override defines a group of override policies about how to override the selected resources and how to apply these resources
+// to the target clusters.
+// Note: override may fail, and it will be reflected on the placement status.
+type Override struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ClusterResourcePlacement.
+	// +required
+	Spec OverrideSpec `json:"spec"`
+}
+
+// OverrideSpec defines the desired state of the Override.
+type OverrideSpec struct {
+	// If none of the ClusterResourceSelectors and ResourceSelectors are specified, it means selecting all resources.
+	// The ClusterResourceSelectors and ResourceSelectors are `ORed`.
+
+	// ClusterResourceSelectors is an array of selectors used to select cluster scoped resources. The selectors are `ORed`.
+	// If a namespace is selected, ALL the resources under the namespace are selected automatically.
+	// You can have 1-100 selectors.
+	// +optional
+	ClusterResourceSelectors []ClusterResourceSelector `json:"clusterResourceSelectors,omitempty"`
```

Comment:
does it still make sense to apply the override to all the resources in a namespace?

Reply:
The main reason is to simplify the duplicate override configuration. For example, a user want to add a label to all of the resources under a namespace. In addition, we can keep it consistent with the CRP semantics.